impl/src/main/java/org/jboss/weld/util/Proxies.java

```diff
@@ -51,13 +54,38 @@
 
     public static class TypeInfo {
 
-        private final Set<Class<?>> interfaces;
-        private final Set<Class<?>> classes;
+        private final List<Class<?>> interfaces;
+        private final List<Class<?>> classes;
 
-        private TypeInfo() {
-            super();
-            this.interfaces = new LinkedHashSet<Class<?>>();
-            this.classes = new LinkedHashSet<Class<?>>();
+        private TypeInfo(Set<? extends Type> types) {
+            Comparator<Class<?>> classComparator = Comparator.comparing(Class::getName);
+            List<Class<?>> foundInterfaces = new ArrayList<>();
+            List<Class<?>> foundClasses = new ArrayList<>();
+
+            types.stream().forEach(type -> add(type, foundInterfaces, foundClasses));
```

Comment:
Just out of curiosity, what's the reason to use a stream and lamda here? ;-)

Reply:
Just to see if you are going to comment on it :-P